### ISSUE TITLE
PHP 7.1 Support

### ISSUE THREAD

**Issue by MathSmath** [aacotroneo/laravel-saml2#122]
This package currently doesn't work using PHP 7.1. The version of the onelogin/php-saml package you're using relies on mcrypt, which has been deprecated for a while, and was completely dropped in 7.1.

onelogin has created a new branch that does not require mcrypt (https://github.com/onelogin/php-saml/tree/remove_mcrypt
), which might be a good approach for this package too. Their 3.0 version will not require mcrypt at all, but I imagine upgrading will require some revisions to your specific implementation of the package. So maybe using their "remove_mcrypt" branch is a good short term fix?

**Comment by danmichaelo** [comment]
I already have such a branch: https://github.com/aacotroneo/laravel-saml2/compare/master...danmichaelo:remove_mcrypt

I don't think I can create a pull request for creating a new branch, but feel free to copy mine.

**Comment by MathSmath** [comment]
Thanks @danmichaelo! Though we were hoping not to have to fork, or rely on someone else's fork :/

**Comment by aacotroneo** [comment]
hi guys! thanks for the collaboration.

I've been away of php for quite a while, don't know how to fix this the
right way (npm peer dependecies don't seem to exist in composer).

I can make a branch of you'd like to have it hosted in this repo until the
main branch evolves

**Comment by MathSmath** [comment]
@aacotroneo That'd be great if you had a branch that referenced the "remove_mcrypt" branch of the php-saml package. https://github.com/onelogin/php-saml/tree/remove_mcrypt

I guess updating to 3.0 might work too, but I don't know when that branch will be stable, or if it'll require any implementation updates on your end.

Thanks man!

**Comment by thomhickey** [comment]
The delta for the remove_mcrypt branch is massive ... are folks using it with this package and having success? It's a choice between possible security problems (see above massive delta) vs. deprecated warnings if I use mcrypt in php 7.1. Hmm ...

**Comment by MathSmath** [comment]
@thomhickey Our app won't even compile in 7.1 because mcrypt is a composer dependency of php-saml. We need to move ahead on several things that require 7.2 (where mcrypt is dropped altogether), so we have little choice unfortunately.

**Comment by thomhickey** [comment]
@MathSmath I see ... and are you using the remove_mcrypt branch with success in production?

**Comment by MathSmath** [comment]
@thomhickey No, that's what this request is about. We could certainly test it out by forking this repo, but would rather avoid maintaining our own forks of packages.

This package (laravel-saml2) is basically incompatible with Laravel 5.6 (which requires php >= 7.1.3) unless there is a version without mcrypt reliant dependencies.

**Comment by danmichaelo** [comment]
I'm using the `remove_mcrypt` branch with success in production. If I remember right, there were a few changes when it came to the defaults for the settings file (keys that could be left out before could not anymore, but had to be defined as blank values or something).

**Comment by aacotroneo** [comment]
Thanks guys! I created a branch with @danmichaelo remove_mcrypt branch 

https://github.com/aacotroneo/laravel-saml2/tree/remove_mcrypt

Unfortunately I don't have any current installation to try it out, but feel free to make PRs. I can grant further permissions if someone wants to maintain the library.

**Comment by thomhickey** [comment]
Ok, going to give the new branch a try thanks @aacotroneo

**Comment by KarimGeiger** [comment]
Just wanted to let you know that we haven't had any problems while upgrading to the remove_mcrypt branch. Still works as expected.

**Comment by nicklee1990** [comment]
@aacotroneo if you let me know what needs doing on the mcrypt branch to get it stable and merged into master I can help out

**Comment by aacotroneo** [comment]
Hi all! Thanks @nicklee1990, I'll really appreciate it because I don't have any environment to test all this anymore.

The main problem is we depend on one login. May be wait for v3.0.0 https://github.com/onelogin/php-saml/issues/255 and make a v1.0.0 release here that uses it.

**Comment by LaravelBen** [comment]
Works perfectly, thank you, although I didn't fork it, I'm just using the existing repo - it was the 'repositories' part I was missing.

**Comment by thomhickey** [comment]
When I implemented it, unless I forked and made the change to the
composer.json file, there was a nasty exception thrown on my /acs route.
Maybe they've updated the repo since then. Anyway, glad it works for you.

**Comment by tjallingt** [comment]
@aacotroneo I'd be willing to help out maintaining the library, at work we have an environment that I can test laravel-saml2 in.

**Comment by aacotroneo** [comment]
that's great @tjallingt! There are a few feature requests, I just try to keep the library simple for 80% of our users and just be a wrapper of one-login. This is an important one though to keep up to date with one-login; still waiting on that release though.

**Comment by rgergo67** [comment]
FYI: php-saml 3.0.0 just released
https://github.com/onelogin/php-saml/issues/255#issuecomment-426339132

**Comment by rroblik** [comment]
Can we expect a master branche release, using https://github.com/onelogin/php-saml/releases/tag/v3.0.0 in a short time delay ?

BR

**Comment by omitobi** [comment]
@aacotroneo Thank you for making this package. As @rgergo67 mentions, onelogin/php-saml has released version 3 of their package that removes the requirement for mcrypt. If anyone is willing to effect this change, it will be great! 🙂

**Comment by perifer** [comment]
Here is a stab at supporting 3.0.0 https://github.com/aacotroneo/laravel-saml2/pull/160

**Comment by aacotroneo** [comment]
thanks- I merged it to a branch so it's easier to centralize changes. when you guys give me a thumbsup I can merge to master and release

**Comment by perifer** [comment]
To test this, change to `dev-one-login-3` in `composer.json`:
```
"require": {
        "aacotroneo/laravel-saml2": "dev-one-login-3",
}
```

And then run `composer update`

**Comment by aacotroneo** [comment]
guys! shall we merge this one? https://github.com/aacotroneo/laravel-saml2/pull/166 It's similar to remove_mcrypt branch, but no dev-ish code.

**Comment by omitobi** [comment]
@aacotroneo Looking good. At least I used that branch `login-3` and it worked.

**Comment by rgergo67** [comment]
Same here, I use one-login-3 and everything works fine.

**Comment by aacotroneo** [comment]
ok! merged to master (v 1.0.0)

**Comment by omitobi** [comment]
Finally In master. Do you think its necessary that the README could be describe about this change a bit especially since this release has major changes on the previous 0.11.1?
